vtm/src/org/oscim/layers/tile/buildings/BuildingLayer.java

```diff
@@ -270,13 +270,41 @@ public ExtrusionRenderer getExtrusionRenderer() {
         return mExtrusionRenderer;
     }
 
+    /**
+     * @return the tile source tag key or library tag key as fallback
+     */
     protected String getKeyOrDefault(String key) {
         if (mTileLayer.getTheme() == null)
             return key;
         String res = mTileLayer.getTheme().transformBackwardKey(key);
         return res != null ? res : key;
     }
 
+    /**
+     * Get the forward transformed value from tile source tag via the library tag key.
+     *
+     * @param key the library tag key
+     * @return the tile source tag value transformed to library tag value
+     */
+    protected String getTransformedValue(MapElement element, String key) {
+        if (mTileLayer.getTheme() == null)
+            return element.tags.getValue(key);
+        key = getKeyOrDefault(key);
+        Tag res = element.tags.get(key);
+        if (res == null) return null;
+        res = mTileLayer.getTheme().transformForwardTag(res);
+        if (res == null) {
+            return element.tags.getValue(key);
+        }
+        return res.value;
+    }
```

Comment:
Please add docs for each step in the method, so that's easily understandable what happens.

Reply:
Updated ;)